rewrite-java/src/main/java/org/openrewrite/java/style/ImportLayoutStyle.java

```diff
@@ -53,12 +118,360 @@
+                        accepted = true;
+                        break;
+                    }
+                }
+            }
+            assert accepted : "Every import must be accepted by at least one block, but this import was not: " + anImport.printTrimmed();
+        }
+
+        int importIndex = 0;
+        String extraLineSpace = "";
+        for (Layout.Block block : layout.getBlocks()) {
+            if (block instanceof Layout.Block.BlankLines) {
+                extraLineSpace = "";
+                for (int i = 0; i < ((Layout.Block.BlankLines) block).getCount(); i++) {
+                    //noinspection StringConcatenationInLoop
+                    extraLineSpace += "\n";
+                }
+            } else {
+                for (J.Import orderedImport : block.orderedImports()) {
+                    String prefix = importIndex == 0 ? originalImports.get(0).getPrefix() :
+                            extraLineSpace + "\n";
+
+                    if (!orderedImport.getPrefix().equals(prefix)) {
+                        orderedImports.add(orderedImport.withPrefix(prefix));
+                    } else {
+                        orderedImports.add(orderedImport);
+                    }
+                    extraLineSpace = "";
+                    importIndex++;
+                }
+            }
+        }
+        return orderedImports;
+    }
 
-        return builder.build();
+    /**
+     * @return Returns a layout style that is modeled after IntelliJ's default import settings.
+     */
+    public static ImportLayoutStyle getDefaultImportLayoutStyle() {
+        return new ImportLayoutStyle();
     }
 
-    @JsonProperty("layout")
-    public void setLayout(Map<String, Object> layout) {
-        this.layout = layout;
+    /**
+     * See {@link ImportLayoutStyle}
+     * <P><P>Block Syntax:<P>
+     *
+     * <PRE>
+     *      "import java.*"                   - All imports in the java package or a subpackage of java
+     *      "import static org.assertj.*"     - All static imports in the "org.assertj" package or subpacakge.
+     *      "&lt;blank line&gt;"                    - A blank line separator
+     *      "import all other imports"        - All other imports
+     *      "import static all other imports" - All other static imports
+     * </PRE>
+     * @param classCountToUseStarImport How many imports from the same package must be present before they should be collapsed into a star import.
+     * @param nameCountToUseStarImport How many static imports from the same type must be present before they should be collapsed into a static star import.
+     * @param blocks An ordered list of import groupings which define exactly how imports should be organized within a compilation unit
+     * @return ImportLayoutStyle
+     */
+    public static @NonNull ImportLayoutStyle layout(int classCountToUseStarImport, int nameCountToUseStarImport, @NonNull  List<String> blocks) {

```

Comment:
To be deserializable from YAML, this will need a `@JsonCreator` and `@JsonProperty` on the params.

Reply:
We should probably have something in our automated testing that ensures that all styles are serializable